views/snippets_api.inc.php

```diff
@@ -0,0 +1,44 @@
+<?php
+namespace Langchecker;
+
+use Transvision\Json;
+
+// This view works only for snippets (website ID 6)
+$current_website = $sites[6];
+$current_locale = $locale;
+$json_data = [];
+
+if ($locale == '') {
+    exit('<p>Missing locale code in the request</p>');
+}
+
+if (! Project::isSupportedLocale($current_website, $current_locale)) {
+    exit('<p>This locale is not supported for snippets</p>');
+}
+
+foreach (Project::getWebsiteFiles($current_website) as $current_filename) {
+    if (! Project::isSupportedLocale($current_website, $current_locale, $current_filename, $langfiles_subsets)) {
+        // File is not managed for this website+locale, ignore it
+        continue;
+    }
+    if (! file_exists(Project::getLocalFilePath($current_website, $current_locale, $current_filename))) {
+        // If the .lang file does not exist, just skip the locale for this file
+        continue;
+    }
+
+    $locale_data = LangManager::loadSource($current_website, $current_locale, $current_filename);
+    foreach ($locale_data['strings'] as $reference_string => $translated_string) {
+        if ($reference_string != $translated_string) {
+            // Interested only in translated strings, clean up {ok}
+            $json_data[$reference_string] = Utils::cleanString($translated_string);
+        }
+    }
+}
+
+ksort($json_data);
+
+if (isset($_GET['callback'])) {
+    echo Json::output($json_data, $_GET['callback'], true);
+} else {
+    echo Json::output($json_data, false, true);
+}
```

Comment:
Simpler I think:
```php
$callback = isset($_GET['callback']) ? $_GET['callback'] : false;
exit(Json::output($json_data, $callback, true));
```
Generally speaking, I prefer to exit when we output json, because we know we have finished processing the request and any other data output afterwards would be a bug.